geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/filter/FilterTest.scala

```diff
@@ -95,7 +96,7 @@ class FilterTester extends Specification with TestWithDataStore with LazyLogging
     val filterCount = mediumDataFeatures.count(filter.evaluate)
     val query = new Query(sftName, filter)
     Option(projection).foreach(query.setPropertyNames)
-    val queryCount = fs.getFeatures(query).size
+    val queryCount = SelfClosingIterator(fs.getFeatures(query)).length
```

Comment:
It may have less to do with returning more correct results than it does correctly disposing of resources once they are no longer needed.  @elahrvivaz ?